File: Kendaraan Practice/Minibus.java
public class Minibus extends Kendaraan{
    Minibus(){
        super();
    }

    // User ctor
    Minibus(int nomor, int tahun, String merk){
        super(nomor, tahun, merk);
    }

    // cctor
    Minibus(Minibus other){
        super(other);
    }

    public Minibus copyUnit(Minibus other){
        super.copyUnit(other);
        return this;
    }

    @Override
    public void printInfo(){
        super.printInfo();
        System.out.printf("Kategori: Minibus%n");
    }

    public int biayaSewa(int lamaSewa){
        float result = lamaSewa<=5 ? (float)5e6 : (float)(5e6+(5e5*(lamaSewa-5)));
        return (int)(result*calculateDiscount(lamaSewa));
    }

    public float calculateDiscount(int lamaSewa)
    {
        return lamaSewa>10? (float)0.9 : (float)1.0 ;
    }

}

File: Kendaraan Practice/Minibus.cpp
#include<bits/stdc++.h>
#include "Minibus.hpp"

using namespace std;

// Default ctor
Minibus::Minibus() : Kendaraan() {}

// User ctor
Minibus::Minibus(int nomor, int tahun, string merk) : Kendaraan(nomor, tahun, merk) {}

// cctor
Minibus::Minibus(const Minibus& other)
{
    this->nomor = other.nomor;
    this->tahun = other.tahun;
    this->merk = other.merk;
}

// dtor
Minibus::~Minibus() {}

void Minibus::printInfo()
{
    Kendaraan::printInfo();
    printf("Kategori: Minibus\n");
}

int Minibus::biayaSewa(int lamaSewa)
{   
    float result = lamaSewa<=5 ? 5e6 : (5e6+(5e5*(lamaSewa-5)));
    return result*calculateDiscount(lamaSewa);
}

float Minibus::calculateDiscount(int lamaSewa)
{
    return lamaSewa>10? 0.9 : 1.0 ;
}